components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/util/SSOAgentUtils.java

```diff
@@ -273,6 +297,56 @@ public static void sendPostResponse(HttpServletRequest request, HttpServletRespo
         }
     }
 
+    /**
+     * Serializing a SAML2 object into a String
+     *
+     * @param xmlObject object that needs to serialized.
+     * @return serialized object
+     * @throws SSOAgentException
+     */
+    public static String marshall(XMLObject xmlObject) throws SSOAgentException {
+        try {
+
+            System.setProperty("javax.xml.parsers.DocumentBuilderFactory",
```

Comment:
Why do we need to set this system property explicitly ? If we are forcefully setting this, don't we need to remove this as well ?